content/influxdb/cloud-dedicated/admin/custom-partitions/define-custom-partitions.md

```diff
@@ -0,0 +1,97 @@
+---
+title: Define custom partitions
+description: >
+  Use the [`influxctl` CLI](/influxdb/cloud-dedicated/reference/cli/influxctl/)
+  to define custom partition strategies when creating a database or measurement.
+menu:
+  influxdb_cloud_dedicated:
+    parent: Manage data partitioning
+weight: 202
+related:
+  - /influxdb/cloud-dedicated/reference/cli/influxctl/database/create/
+---
+
+Use the [`influxctl` CLI](/influxdb/cloud-dedicated/reference/cli/influxctl/)
+to define custom partition strategies when creating a database or measurement.
+By default, {{< product-name >}} partitions data by day.
+
+The partition strategy of a database or measurement is determined by a
+_partition template_ which defines the naming pattern for _partition keys_.
+Partition keys uniquely identify each partition.
+When a partition template is applied to a database, it becomes the default template
+for all measurements in that database, but can be overridden when creating a
+measurement.
+
+- [Create a database with a custom partition template](#create-a-database-with-a-custom-partition-template)
+- [Create a measurement with a custom partition template](#create-a-measurement-with-a-custom-partition-template)
+- [Example partition templates](#example-partition-templates)
+
+{{% note %}}
+#### Partition templates can only be applied on create
+
+You can only apply a partition template when creating a database or measurement.
+There is no way to update a partition template on an existing resource.
+{{% /note %}}
+
+Use the following command flags to identify
+[partition template parts](/influxdb/cloud-dedicated/admin/custom-partitions/partition-templates/#tag-part-templates):
+
+- `--template-tag`: An [InfluxDB tag](/influxdb/cloud-dedicated/reference/glossary/#tag)
+  to use in the partition template.
+  _Supports up to 7 of these flags._
+- `--template-time`: A [Rust strftime date and time](/influxdb/cloud-dedicated/admin/custom-partitions/partition-templates/#time-part-templates)
+  string that specifies the time format in the partition template and determines
+  the time interval to partition by.
+
+_View [partition template part restrictions](/influxdb/cloud-dedicated/admin/custom-partitions/partition-templates/#restrictions)._
+  
+## Create a database with a custom partition template
+
+The following example creates a new `example-db` database and applies a partition
+template that partitions by two tags (`room` and `sensor-type`) and by hour using
+the time format `%Y-%m-%d %H:00`:
+
+```sh
+influxctl database create \
+  --template-tag room \
+  --template-tag sensor-type \
+  --template-time '%Y-%m-%d %H:00' \
+  example-db
+```
+
+## Create a measurement with a custom partition template
+
+The following example creates a new `example-meas` measurement and applies a
+partition template that partitions by two tags (`room` and `sensor-type`) and by
+week using the time format `%Y-%m-%d %H:00`:
+
+```sh
+influxctl measurement create \
```

Comment:
Hmm I had assumed that I would call this `tables` instead of `measurement`. When creating, listing, and updating databases today with `influxctl`, we have `max tables` and `max columns per table` as options and in the output. I am inclined to stick with table to be consistent throughout.

I even see `Use SHOW TABLES to list measurements in your InfluxDB database.` in the docs, which feels like we are combining two terms.

Reply:
Yeah, I had a feeling you were leaning this way. This brings up an interesting discussion that is honestly, probably bigger than this PR.

When describing the InfluxDB data structure, we still use "databases," "measurements," "fields," and "tags" which are all inherited from previous versions of InfluxDB and existing users have an established mental model based on these. With IOx and the first class citizen status of SQL, under the hood, we now use the traditional columnar data model (tables, columns, etc.) where TSM... wasn't that. Yes, under the hood, a "measurement" is represented by table and fields and tags are just columns in a table, but to users, we still push the measurement/tag/field model. So I guess the bigger question is--should we? 🤷🏻‍♂️

> I even see `Use SHOW TABLES to list measurements in your InfluxDB database.` in the docs, which feels like we are combining two terms.

This is specific to SQL and uses the metaqueries shipped with DataFusion. With InfluxQL, you'd use [`SHOW MEASUREMENTS`](https://docs.influxdata.com/influxdb/cloud-serverless/reference/influxql/show/#show-measurements).

I don't have the answer for this and I don't think you do either.

Maybe in the mean time, and as a middle-ground, `influxctl measurement` could be an alias of `influxctl table`?

Reply:
> I don't have the answer for this and I don't think you do either.

Who is the decision maker?

> Maybe in the mean time, and as a middle-ground, influxctl measurement could be an alias of influxctl table?

We should have a preferred way of doing things and drive our users towards that. Adding two ways to do the same thing is something I would strongly like to avoid as the feedback from previous CLIs was the complexity and verbosity made them difficult to use.

Reply:
> Who is the decision maker?

I don't know who exactly would be, but I think @garylfowler, @barbaranelson, and @pauldix may want to weigh in (going right to the top 😄).

> We should have a preferred way of doing things and drive our users towards that. Adding two ways to do the same thing is something I would strongly like to avoid as the feedback from previous CLIs was the complexity and verbosity made them difficult to use.

I agree. My preference would be `measurement` since that's what we use _almost_ everywhere else (line protocol, data model descriptions, schemas, etc.).

Reply:
My second preference would be table, since that is going to be a known term for people using us for the first time but familiar with other database.  But my first preference would be to allow for either (table or measurement) since we do talk about measurement still in a lot of places in our Docs, and certainly when we talk about line protocol and we also have customers migrating from previous versions.

Reply:
With 3.0 the preferred naming convention is database, tables, columns, rows. This will matter even more when we get to the 3.0 native data model and new write API.

I'd generally prefer not to use measurement as an alias unless we're talking specifically about line protocol or InfluxQL.

It's better to make it clear to the user that 3.0 has the model of organizing data into database > table > partition > parquet files and that we have a way to map 1.x and 2.x schema onto that model. But the sooner they start thinking about the model in the 3.0 way, the closer they'll be to creating a schema that will yield good results.